### ISSUE TITLE
(+FIXED) CS:GO Crashes on the New Game Update as of 09/20/2019 + (New Offsets Change)

### ISSUE THREAD

**Issue by VuBrian22** [danielkrupinski/Osiris#542]
Hopefully this gets fixed soon.....PS for anyone who wants to update it and make a pull request, you can find the new offsets here: https://github.com/frk1/hazedumper/ EDIT: The new offsets do not work. You have to change the signature in Memory.cpp. Credit to deprae and wolf2000.

**wolf2000: (FIX)**

+++ b/Osiris/SDK/Entity.h
@@ -103,7 +103,7 @@ public:
     constexpr Vector getEyePosition() noexcept
     {
         Vector vec{ };
-        callVirtualMethod<void, Vector&>(this, 281, vec);
+        callVirtualMethod<void, Vector&>(this, 282, vec);
         return vec;
     }

@@ -142,7 +142,7 @@ public:

     constexpr bool isWeapon() noexcept
     {
-        return callVirtualMethod<bool>(this, 163);
+        return callVirtualMethod<bool>(this, 164);
     }

     constexpr ClientClass* getClientClass() noexcept
@@ -152,32 +152,32 @@ public:

     constexpr bool isAlive() noexcept
     {
-        return callVirtualMethod<bool>(this, 153) && health() > 0;
+        return callVirtualMethod<bool>(this, 154) && health() > 0;
     }

     constexpr bool isPlayer() noexcept
     {
-        return callVirtualMethod<bool>(this, 155);
+        return callVirtualMethod<bool>(this, 156);
     }

     constexpr Entity* getActiveWeapon() noexcept
     {
-        return callVirtualMethod<Entity*>(this, 265);
+        return callVirtualMethod<Entity*>(this, 266);
     }

     constexpr int getWeaponSubType() noexcept
     {
-        return callVirtualMethod<int>(this, 277);
+        return callVirtualMethod<int>(this, 278);
     }

     constexpr WeaponData* getWeaponData() noexcept
     {
-        return callVirtualMethod<WeaponData*>(this, 454);
+        return callVirtualMethod<WeaponData*>(this, 455);
     }

     constexpr float getInaccuracy() noexcept
     {
-        return callVirtualMethod<float>(this, 476);
+        return callVirtualMethod<float>(this, 477);
     }

     VarMap* getVarMap() noexcept
@@ -215,7 +215,7 @@ public:

     constexpr Entity* getObserverTarget() noexcept
     {
-        return callVirtualMethod<Entity*>(this, 291);
+        return callVirtualMethod<Entity*>(this, 292);
     }

     bool isInReload() noexcept

deprae:

 "The problem is that VALVE updated client_panorama.dll which in turn broke all the signatures.
You either reverse the stuff needed and find the signatures urself or wait for daniel to do it.
To find out what you have to update just debug and crash and update the signatures needed.
Chances are most of signatures are broken.

Signatures = / = Offsets.

Signatures = More update proof. You basically make a signature out of HEX data.
Offsets = Updated on each game update, a pain in the fucking ass."

**Comment by deprale** [comment]
Can confirm.

**Comment by LuanGraa** [comment]
**_### Every_**  cheat is crashing rn because valve changed the offsets.

**Comment by VuBrian22** [comment]
> **_### Every_** cheat is crashing rn because valve changed the offsets.

Yeah I know that. I just opened this issue because it acts as a "red flag" for anyone who wants to update the offsets. I could do it myself and make a pull request, however I have 3 exams tomorrow.

**Comment by LuanGraa** [comment]
No problem bud, i would fix it if i knew how to but i am just a dirty paster.

**Comment by deprale** [comment]
https://i.imgur.com/1iCNm3q.png

**Comment by LuanGraa** [comment]
Excuse my ignorance but what is exactly the problem then?

**Comment by VuBrian22** [comment]
> Excuse my ignorance but what is exactly the problem then?

Basically to make it work again, you have to change the signatures in Memory.cpp instead of using the offsets from hazedumper because the cheat uses interfaces. And the dumped offsets will not work  with this cheat.

**Comment by deprale** [comment]
The problem is that VALVE updated client_panorama.dll which in turn broke all the signatures.
You either reverse the stuff needed and find the signatures urself or wait for daniel to do it.
To find out what you have to update just debug and crash and update the signatures needed.
Chances are most of signatures are broken.

Signatures = / = Offsets.

Signatures = More update proof. You basically make a signature out of HEX data.
Offsets = Updated on each game update, a pain in the fucking ass.

**Comment by LuanGraa** [comment]
Ah ok i see thanks, i will wait for the homeboy daniel then because i cant use WinHEX for shit.

**Comment by deprale** [comment]
@danielkrupinski  and everyone else in this thread
Some good news from this update <3
https://i.imgur.com/PST7Evd.png
You know what this means? Basically CS:GO devs made us a free grenade trajectory LUL.
Cast a nullptr to that biiih, free grenade trajectory LULw. EASY!

**Comment by LuanGraa** [comment]
Oh so no math needed anymore?, sick

**Comment by VuBrian22** [comment]
> @danielkrupinski and everyone else in this thread
> Some good news from this update <3
> https://i.imgur.com/PST7Evd.png
> You know what this means? Basically CS:GO devs made us a free grenade trajectory LUL.
> Cast a nullptr to that biiih, free grenade trajectory LULw. EASY!

OML Here comes the opportunity to exploit the feature. Lmao.

**Comment by deprale** [comment]
Best part is that this shit can be done in like 2 lines total - the void statement and the if config statement LULw

**Comment by LuanGraa** [comment]
Another ignorant comment here, would it be able to set this variable to true using FindVar and set Value?

**Comment by deprale** [comment]
not to true but just 1
but first you'd have to cast a nullptr to it in hooks

**Comment by deprale** [comment]
@LuanGraa https://i.imgur.com/z353Uya.png
see last 3 lines for example.

**Comment by LuanGraa** [comment]
Ah ok thanks for the info

**Comment by mdilai** [comment]
So much offtop and nobody post a fix...

```--- a/Osiris/SDK/Entity.h
+++ b/Osiris/SDK/Entity.h
@@ -103,7 +103,7 @@ public:
     constexpr Vector getEyePosition() noexcept
     {
         Vector vec{ };
-        callVirtualMethod<void, Vector&>(this, 281, vec);
+        callVirtualMethod<void, Vector&>(this, 282, vec);
         return vec;
     }

@@ -142,7 +142,7 @@ public:

     constexpr bool isWeapon() noexcept
     {
-        return callVirtualMethod<bool>(this, 163);
+        return callVirtualMethod<bool>(this, 164);
     }

     constexpr ClientClass* getClientClass() noexcept
@@ -152,32 +152,32 @@ public:

     constexpr bool isAlive() noexcept
     {
-        return callVirtualMethod<bool>(this, 153) && health() > 0;
+        return callVirtualMethod<bool>(this, 154) && health() > 0;
     }

     constexpr bool isPlayer() noexcept
     {
-        return callVirtualMethod<bool>(this, 155);
+        return callVirtualMethod<bool>(this, 156);
     }

     constexpr Entity* getActiveWeapon() noexcept
     {
-        return callVirtualMethod<Entity*>(this, 265);
+        return callVirtualMethod<Entity*>(this, 266);
     }

     constexpr int getWeaponSubType() noexcept
     {
-        return callVirtualMethod<int>(this, 277);
+        return callVirtualMethod<int>(this, 278);
     }

     constexpr WeaponData* getWeaponData() noexcept
     {
-        return callVirtualMethod<WeaponData*>(this, 454);
+        return callVirtualMethod<WeaponData*>(this, 455);
     }

     constexpr float getInaccuracy() noexcept
     {
-        return callVirtualMethod<float>(this, 476);
+        return callVirtualMethod<float>(this, 477);
     }

     VarMap* getVarMap() noexcept
@@ -215,7 +215,7 @@ public:

     constexpr Entity* getObserverTarget() noexcept
     {
-        return callVirtualMethod<Entity*>(this, 291);
+        return callVirtualMethod<Entity*>(this, 292);
     }

     bool isInReload() noexcept

```

**Comment by deprale** [comment]
@mdilai 
thx for fix, mind explaining how you figured out? 

share some knowledge lul

**Comment by LuanGraa** [comment]
thanks dude real good shit.

**Comment by mdilai** [comment]
> @mdilai
> thx for fix, mind explaining how you figured out?
> 
> share some knowledge lul

I went on https://www.unknowncheats.me/forum/counterstrike-global-offensive/103220-counterstrike-global-offensive-reversal-structs-offsets.html and read the last two pages where found most of new offsets. Quickly figured that offsets just incremented by 1 so I've incremented all offsets except of `getAbsOrigin()` and `setModelIndex()` coz backtrack and skinchanger _looks_ still working and i made assumption that some new function (which shifted all other functions right) _probably_ added after "setModelIndex()"

**Comment by VuBrian22** [comment]
> > @mdilai
> > thx for fix, mind explaining how you figured out?
> > share some knowledge lul
> 
> I went on https://www.unknowncheats.me/forum/counterstrike-global-offensive/103220-counterstrike-global-offensive-reversal-structs-offsets.html and read the last two pages where found most of new offsets. Quickly figured that offsets just incremented by 1 so I've incremented all offsets except of `getAbsOrigin()` and `setModelIndex()` coz backtrack and skinchanger _looks_ still working and i made assumption that some new function (which shifted all other functions right) _probably_ added after "setModelIndex()"

To make things easier, would you mind making a pull request after you test it?

**Comment by LuanGraa** [comment]
shieeeet just got untrusted ban.

**Comment by mdilai** [comment]
> shieeeet just got vac authentication error.

Oops, sorry dude if this is my fault. Are you used skinchanger? If yes, probably setModelIndex() is incremented too because it is the only function which change something, rest are just read values from client.

**Comment by Bywnb** [comment]
@danielkrupinski

**Comment by LuanGraa** [comment]
Nah it was my fucking injector, sorry to worry you guys the cheat is probably fine

**Comment by VuBrian22** [comment]
This works now lol!

**Comment by 0xE232FE** [comment]
VTable Index vs Offsets who will ever win this fight in this Issue Thread. Guys this is so funny that no one knows what all are talking about. Just learn coding before posting bullshit here. Many in this thread don't even know what an offset is and copy & paste things from hazedump... Guys please try to learn about Offsets and look into the solution what is VTable Index. So you have two phrases where you can start digging and learn. Offsets and VTable Index search for it you will get illumniated fast!

**Comment by VuBrian22** [comment]
> VTable Index vs Offsets who will ever win this fight in this Issue Thread. Guys this is so funny that no one knows what all are talking about. Just learn coding before posting bullshit here. Many in this thread don't even know what an offset is and copy & paste things from hazedump... Guys please try to learn about Offsets and look into the solution what is VTable Index. So you have two phrases where you can start digging and learn. Offsets and VTable Index search for it you will get illumniated fast!

That is what Github is for. This is a learning experience for all of us. We learn together.

**Comment by wolff20000** [comment]
> So much offtop and nobody post a fix...
> 
> ```
> +++ b/Osiris/SDK/Entity.h
> @@ -103,7 +103,7 @@ public:
>      constexpr Vector getEyePosition() noexcept
>      {
>          Vector vec{ };
> -        callVirtualMethod<void, Vector&>(this, 281, vec);
> +        callVirtualMethod<void, Vector&>(this, 282, vec);
>          return vec;
>      }
> 
> @@ -142,7 +142,7 @@ public:
> 
>      constexpr bool isWeapon() noexcept
>      {
> -        return callVirtualMethod<bool>(this, 163);
> +        return callVirtualMethod<bool>(this, 164);
>      }
> 
>      constexpr ClientClass* getClientClass() noexcept
> @@ -152,32 +152,32 @@ public:
> 
>      constexpr bool isAlive() noexcept
>      {
> -        return callVirtualMethod<bool>(this, 153) && health() > 0;
> +        return callVirtualMethod<bool>(this, 154) && health() > 0;
>      }
> 
>      constexpr bool isPlayer() noexcept
>      {
> -        return callVirtualMethod<bool>(this, 155);
> +        return callVirtualMethod<bool>(this, 156);
>      }
> 
>      constexpr Entity* getActiveWeapon() noexcept
>      {
> -        return callVirtualMethod<Entity*>(this, 265);
> +        return callVirtualMethod<Entity*>(this, 266);
>      }
> 
>      constexpr int getWeaponSubType() noexcept
>      {
> -        return callVirtualMethod<int>(this, 277);
> +        return callVirtualMethod<int>(this, 278);
>      }
> 
>      constexpr WeaponData* getWeaponData() noexcept
>      {
> -        return callVirtualMethod<WeaponData*>(this, 454);
> +        return callVirtualMethod<WeaponData*>(this, 455);
>      }
> 
>      constexpr float getInaccuracy() noexcept
>      {
> -        return callVirtualMethod<float>(this, 476);
> +        return callVirtualMethod<float>(this, 477);
>      }
> 
>      VarMap* getVarMap() noexcept
> @@ -215,7 +215,7 @@ public:
> 
>      constexpr Entity* getObserverTarget() noexcept
>      {
> -        return callVirtualMethod<Entity*>(this, 291);
> +        return callVirtualMethod<Entity*>(this, 292);
>      }
> 
>      bool isInReload() noexcept
> ```
how i can edit the dll for fix it?

**Comment by 0xE232FE** [comment]
@wolff20000 edit this file **Osiris/SDK/Entity.h** and recompile!

**Comment by effex1337** [comment]
![image](https://user-images.githubusercontent.com/48663440/65305583-4e38ac80-dba3-11e9-826c-1cbc0768bccc.png)
hmm idk why its crashing here
🤔

**Comment by effex1337** [comment]
![image](https://user-images.githubusercontent.com/48663440/65305701-8e982a80-dba3-11e9-9a1a-443953286950.png)

**Comment by VuBrian22** [comment]
@effex1337 why are you changing hooks.cpp? And the code couldn't compile because you just put all the code you had screenshot ed in C++ comments?

**Comment by effex1337** [comment]
OMG!
i commented it cause it crashed there
my osiris src has 2x more stuff
`why are you changing hooks.cpp? `
cause it crashed there?, cause added new hooks?
and i already fixed

**Comment by tirziz** [comment]
One thing good in this update, 

> @danielkrupinski and everyone else in this thread
> Some good news from this update <3
> https://i.imgur.com/PST7Evd.png
> You know what this means? Basically CS:GO devs made us a free grenade trajectory LUL.
> Cast a nullptr to that biiih, free grenade trajectory LULw. EASY!

wow. no more delay on grenade trajectory . this is good

**Comment by mdilai** [comment]
Who the hell is wolf2000 and why my poorly pasted solution from https://github.com/danielkrupinski/Osiris/issues/542#issuecomment-533390664 credited to him?

**Comment by wolff20000** [comment]
I don’t know why too, maybe they quote wrong, they gonna fix it.

**Comment by mdilai** [comment]
No problem, dude. I just spent few hours this night to figure this out and share it with community, so get a bit frustrated when I saw that my work was appropriated.
By the way, i've played about 6 hours after fix and did not get any crash at all, so i really doubt that we need to change any signature in Memory.cpp. And hazedumper completely useless in this case, it dumps pointer offsets but we need vtable indexes for Osiris (and generally it mostly for external cheats). So opening post is absolutely incorrect.
